models/users.js

```diff
@@ -306,6 +309,78 @@ const initializeUser = async (userId) => {
   return true;
 };
 
+/**
+ * Adds user data for verification by moderators
+ * @param userId {String} - RDS User Id
+ * @param discordId {Object} - Discord id of RDS user
+ * @param profileImageUrl {number} - profile image URL of user
+ * @param discordImageUrl {number} - discord image URL of user
+ */
+const addForVerification = async (userId, discordId, profileImageUrl, discordImageUrl) => {
+  try {
+    const isNotVerified = await photoVerificationModel.where("userId", "==", userId).limit(1).get();
+    const unverifiedUserData = {
+      userId,
+      discordId,
+      discord: { url: discordImageUrl, approved: false, date: admin.firestore.Timestamp.fromDate(new Date()) },
+      profile: { url: profileImageUrl, approved: false, date: admin.firestore.Timestamp.fromDate(new Date()) },
+    };
+    if (!isNotVerified.empty) {
+      const documentRef = isNotVerified.docs[0].ref;
+      // DOESN"T CHANGE THE APPROVAL STATE OF DISCORD IMAGE IF ALREADY VERIFIED
+      unverifiedUserData.discord.approved = isNotVerified.docs[0].data().discord.approved || false;
+      // DOESN"T CHANGE THE UPDATE DATE OF DISCORD IMAGE
+      unverifiedUserData.discord.date =
+        isNotVerified.docs[0].data().discord.date || admin.firestore.Timestamp.fromDate(new Date());
+      await documentRef.update(unverifiedUserData);
+    } else await photoVerificationModel.add(unverifiedUserData);
+    return { message: "Profile data added for verification successfully" };
+  } catch (err) {
+    logger.error("Error in creating Verification Entry", err);
+    throw err;
+  }
+};
+
+/**
+ * Removes if user passed a valid image; ignores if no unverified record
+ * @param userId {String} - RDS user Id
+ * @param type {String} - type of image that was verified
+ */
+const markAsVerified = async (userId, imageType) => {
+  try {
+    const verificationUserDataSnapshot = await photoVerificationModel.where("userId", "==", userId).limit(1).get();
+    // THROWS ERROR IF NO DOCUMENT FOUND
+    if (verificationUserDataSnapshot.empty) {
+      throw new Error("No verification document record data for user was found");
+    }
+    // VERIFIES BASED ON THE TYPE OF IMAGE
+    const imageVerificationType = imageType === "discord" ? "discord.approved" : "profile.approved";
+    const documentRef = verificationUserDataSnapshot.docs[0].ref;
+    await documentRef.update({ [imageVerificationType]: true });
+    return { message: "User image data verified successfully" };
+  } catch (err) {
+    logger.error("Error while Removing Verification Entry", err);
+    throw err;
+  }
+};
+
+/**
+ * Removes if user passed a valid image; ignores if no unverified record
+ * @param userId {String} - RDS user Id
+ */
+const getUserImageForVerification = async (userId) => {
+  try {
+    const verificationImagesSnapshot = await photoVerificationModel.where("userId", "==", userId).limit(1).get();
```

Comment:
Instead of using the try-catch block around the current lines, it can be applied specifically around this line.